### ISSUE TITLE
Noticed a lot of code tends to use linq

### ISSUE THREAD

**Issue by emrys90** [alelievr/NodeGraphProcessor#130]
Linq allocates memory on each call, so it's not ideal. A lot of the link and delegate code could be changed to not allocate memory.

**Comment by alelievr** [comment]
Yes, I know. I mainly use them in Editor or views because there is already so much allocation in the unity editor itself (and the functions that draw fields, etc.) that nobody cares anymore 🤷 

But for the runtime processing, it should be very light in terms of allocations. Did you have any kind of issue related to GC allocs in NodeGraphProcessor?

**Comment by emrys90** [comment]
I haven't ran into any issues yet as I am still implementing it into my project, but I believe I will due to the number of graphs I need to run. I'm working on a VR game, so any dropped frames is very bad. An example of a runtime allocation I found is NodePorts.PullDatas, that could be a foreach loop with no allocations instead
